source/common/router/rds_impl.cc

```diff
@@ -174,5 +185,45 @@ Router::RouteConfigProviderSharedPtr RouteConfigProviderManagerImpl::getRouteCon
   return new_provider;
 };
 
+void RouteConfigProviderManagerImpl::addRouteInfo(const RdsRouteConfigProvider& provider,
+                                                  Buffer::Instance& response) {
+  response.add("{\n");
+  response.add(fmt::format("    \"route_config_name\": \"{}\",\n", provider.routeConfigName()));
+  response.add(fmt::format("    \"cluster_name\": \"{}\",\n", provider.clusterName()));
+  response.add("    \"route_table_dump\": ");
+  response.add(fmt::format("{}\n", provider.configAsJson()));
+  response.add("}");
+}
+
+Http::Code RouteConfigProviderManagerImpl::handlerRoutes(const std::string& url,
+                                                         Buffer::Instance& response) {
+  Http::Utility::QueryParams query_params = Http::Utility::parseQueryString(url);
+  // If there are no query params, print out all the configured route tables.
+  if (query_params.size() == 0) {
+    for (const auto& provider : rdsRouteConfigProviders()) {
+      addRouteInfo(*provider, response);
+    }
+    return Http::Code::OK;
+  }
+
+  // If there are query params, make sure it is only the route_config_name param.
+  const auto it = query_params.find("route_config_name");
+  if (query_params.size() == 1 && it != query_params.end()) {
+    for (const auto& provider : rdsRouteConfigProviders()) {
+      if (provider->routeConfigName() == it->second) {
+        addRouteInfo(*provider, response);
+      }
+    }
+    return Http::Code::OK;
+  }
+  response.add("{\n");
```

Comment:
Here's an idea that is out-of-scope for this PR, but might be worthy of a TODO or issue filing. How about we define all the handler output formats as protos in `envoy-api`, and just use the JSON transcoding here? This would avoid having to build JSON object by hand and provide the general protoey goodness.